src/main/kotlin/io/github/anki/anki/controller/ControllerExceptionHandler.kt

```diff
@@ -1,24 +1,43 @@
 package io.github.anki.anki.controller
 
+import io.github.anki.anki.service.exceptions.CardDoesNotExistException
+import io.github.anki.anki.service.exceptions.DeckDoesNotExistException
+import io.github.anki.anki.service.exceptions.DoesNotExist
 import org.slf4j.LoggerFactory
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
+import org.springframework.web.HttpRequestMethodNotSupportedException
+import org.springframework.web.bind.MethodArgumentNotValidException
 import org.springframework.web.bind.annotation.ExceptionHandler
-import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestControllerAdvice
 
 @RestControllerAdvice
 class ControllerExceptionHandler {
+    @ExceptionHandler(HttpRequestMethodNotSupportedException::class)
+    fun methodNotSupportedHandler(): ResponseEntity<String> =
+        ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not Found")
+
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleValidationExceptions(ex: MethodArgumentNotValidException): ResponseEntity<Map<String, String>> =
+        ResponseEntity(ex.toMap(), HttpStatus.BAD_REQUEST)
+
+    @ExceptionHandler(DeckDoesNotExistException::class, CardDoesNotExistException::class)
+    fun doesNotExistHandler(ex: DoesNotExist): ResponseEntity<String> =
+        ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.message)
 
     @ExceptionHandler(Exception::class)
-    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
     fun globalExceptionHandler(
         ex: Exception,
     ): ResponseEntity<String> {
         LOG.error("Handling global exception", ex)
         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Internal Server Error")
     }
 
+    fun MethodArgumentNotValidException.toMap(): Map<String, String> {
```

Comment:
private? 
if it used not in private, move it to `controller.mapper` package (not to `dto.mapper`)

Reply:
ok i will move it to controller.mapper

Reply:
oh no, i deleted test for this yesterday so i will make it private